extlinks/common/helpers.py

```diff
@@ -52,11 +52,14 @@ def get_linksearchtotal_data_by_time(queryset, start_date=None, end_date=None):
             current_date = month_first - timedelta(days=1)
 
         # If a month has no data for some reason, we should use whatever
-        # figure we have for the previous month, unless it is the current month
+        # figure we have for the previous month
         for i, data in enumerate(linksearch_data):
-            if data == 0 and i != len(linksearch_data) - 1:
-                linksearch_data[i] = linksearch_data[i + 1]
-
+            linksearch_data_length = len(linksearch_data)
+            if data == 0 and i != linksearch_data_length - 1:
+                for j in range(i + 1, linksearch_data_length):
```

Comment:
I think we discussed it in a meeting, but we should also change the interface if the current total isn't the most recent month. Let's change the UI as well.